docs/apis/core/dml/index.md

```diff
@@ -0,0 +1,1066 @@
+
+A general way to iterate over records using the get_recordset_xxx() functions:
+
+```php
+$rs = $DB->get_recordset(....);
+foreach ($rs as $record) {
+    // Do whatever you want with this record
+}
+$rs->close();
+```
+
+Unlike get_record functions, you cannot check if <tt>$rs  = = true</tt> or <tt>!empty($rs)</tt> to determine if any records were found. Instead, if you need to, you can use:
+
+```php
+if ($rs->valid()) {
+    // The recordset contains some records.
+}
+```
+
+### get_recordset
+
+Return a list of records as a moodle_recordset where all the given conditions are met.
+
+```php
+public function get_recordset(
+    $table,
+    array $conditions = null,
+    $sort = '',
+    $fields = '*',
+    $limitfrom = 0,
+    $limitnum = 0
+);
+```
+
+### get_recordset_select
+
+Return a list of records as a moodle_recordset where the given conditions are used in the WHERE clause.
+
+```php
+public function get_recordset_select(
+    $table,
+    $select,
+    array $params = null,
+    $sort = '',
+    $fields = '*',
+    $limitfrom = 0,
+    $limitnum = 0
+);
+```
+
+### get_recordset_sql
+
+Return a list of records as a moodle_recordset using a custom SELECT query.
+
+```php
+public function get_recordset_sql(
+    $sql,
+    array $params = null,
+    $limitfrom = 0,
+    $limitnum = 0
+);
+```
+
+### get_recordset_list
+
+Return a list of records as a moodle_recordset where the given field matches one of the possible values.
+
+```php
+public function get_recordset_list(
+    $table,
+    $field,
+    array $values,
+    $sort = *,
+    $fields = '*',
+    $limitfrom = *,
+    $limitnum = ''
+);
+```
+
+## Delegated transactions
+
+- Please note some databases do not support transactions (such as the MyISAM MySQL database engine), however all server administrators are strongly encouraged to migrate to databases that support transactions (such as the InnoDB MySQL database engine).
+- Previous versions supported only one level of transaction. Since Moodle 2.0, the DML layer emulates delegated transactions that allow nesting of transactions.
+- Some subsystems (such as messaging) do not support transactions because it is not possible to rollback in external systems.
+A transaction is started by:
+
+```php
+$transaction = $DB->start_delegated_transaction();
+```
+
+and finished by:
+
+```php
+$transaction->allow_commit();
+```
+
+Usually a transaction is rolled back when an exception is thrown:
+
+```php
+$transaction->rollback($ex);
+```
+
+which must be used very carefully because it might break compatibility with databases that do not support transactions. Transactions cannot be used as part of expected code flow; they can be used only as an emergency protection of data consistency.
+
+See more details in [DB layer 2.0 delegated transactions](https://docs.moodle.org/dev/DB_layer_2.0_delegated_transactions) or [MDL-20625](https://tracker.moodle.org/browse/MDL-20625).
```

Comment:
```suggestion
:::info More information:

See more details in [DB layer 2.0 delegated transactions](https://docs.moodle.org/dev/DB_layer_2.0_delegated_transactions) or [MDL-20625](https://tracker.moodle.org/browse/MDL-20625).

:::
```

Reply:
I've also created https://github.com/moodle/devdocs/issues/217 with a proposal to migrate this page (adding the `good first issue` label too)